qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -59,21 +63,39 @@ def __init__(
                 Only enforced when not using the state vector simulator. Default True.
             batch_size: Number of circuits to batch together for computation. Default 1000.
             quantum_instance: Quantum Instance or Backend
+            user_parameters: Iterable containing Parameter objects which correspond to
+                 quantum gates on the feature map circuit which may be tuned. If users intend to
+                 tune feature map parameters to find optimal values, this field should be set.
         """
-        self._feature_map = feature_map if feature_map else ZZFeatureMap(2)
+        # Class fields
+        self._feature_map = None
+        self._unbound_feature_map = None
+        self._user_parameters = None
+        self._user_param_binds = None
         self._enforce_psd = enforce_psd
         self._batch_size = batch_size
         self._quantum_instance = quantum_instance
 
+        self.feature_map = feature_map if feature_map else ZZFeatureMap(2)
+        self.user_parameters = user_parameters
```

Comment:
I think these declarations are not required since we have kind of private class members (`_feature_map`, `_user_parameters` and property setters/getters.

Reply:
Each of these  setters contains a good amount of logic, so I kept these lines in here to keep the constructor clean.

Reply:
> 
> Each of these setters contains a good amount of logic, so I kept these lines in here to keep the constructor clean.

Ah, I see now, I forgot about setters. You're right.